pantheon-sessions.php

```diff
@@ -261,6 +261,206 @@ public static function force_first_load() {
+		$key_existence = $wpdb->get_results( $query );
+
+		// Avoid errors by not attempting to add a column that already exists.
+		if ( ! empty( $key_existence ) ) {
+			$this->safe_output( __( 'ID column already exists and does not need to be added to the table.', 'wp-native-php-sessions' ), 'error' );
+		}
+
+		// Alert the user that the action is going to go through.
+		$this->safe_output( __( 'Primary Key does not exist, resolution starting.', 'wp-native-php-sessions' ), 'log' );
+
+		$count_query = "SELECT COUNT(*) FROM {$table};";
+		$count_total = $wpdb->get_results( $count_query );
+		$count_total = $count_total[0]->{'COUNT(*)'};
+
+		if ( $count_total >= 20000 ) {
+			// translators: %s is the total number of rows that exist in the pantheon_sessions table.
+			$this->safe_output( __( 'A total of %s rows exist. To avoid service interruptions, this operation will be run in batches. Any sessions created between now and when operation completes may need to be recreated.', 'wp-native-php-sessions' ), 'log', [ $count_total ] );
+		}
+		// Create temporary table to copy data into in batches.
+		$query = "CREATE TABLE {$temp_clone_table} LIKE {$table};";
+		$wpdb->query( $query );
+		$query = "ALTER TABLE {$temp_clone_table} ADD COLUMN id BIGINT AUTO_INCREMENT PRIMARY KEY FIRST";
+		$wpdb->query( $query );
+
+		$batch_size = 20000;
+		$loops      = ceil( $count_total / $batch_size );
+
+		for ( $i = 0; $i < $loops; $i++ ) {
+			$offset = $i * $batch_size;
+
+			$query           = sprintf( "INSERT INTO {$temp_clone_table} 
+(user_id, session_id, secure_session_id, ip_address, datetime, data) 
+SELECT user_id,session_id,secure_session_id,ip_address,datetime,data 
+FROM %s ORDER BY user_id LIMIT %d OFFSET %d", $table, $batch_size, $offset );
+			$results         = $wpdb->query( $query );
+			$current_results = $results + ( $batch_size * $i );
+
+			// translators: %1 and %2 are how many rows have been processed out of how many total.
+			$this->safe_output( __( 'Updated %1$s / %2$s rows. ', 'wp-native-php-sessions' ), 'log', [ $current_results, $count_total ] );
+		}
+
+		// Hot swap the old table and the new table, deleting a previous old
+		// table if necessary.
+		$old_table = $wpdb->base_prefix . 'bak_' . $unprefixed_table;
+		$query     = $wpdb->prepare( 'SHOW TABLES LIKE %s', $wpdb->esc_like( $old_table ) );
+
+		if ( $wpdb->get_var( $query ) == $old_table ) {
+			$query = "DROP TABLE {$old_table};";
+			$wpdb->query( $query );
+		}
+
+		$query = "ALTER TABLE {$table} RENAME {$old_table};";
+		$wpdb->query( $query );
+		$query = "ALTER TABLE {$temp_clone_table} RENAME {$table};";
+		$wpdb->query( $query );
+
+		$this->safe_output( __( 'Operation complete, please verify that your site is working as expected. When ready, run terminus wp {site_name}.{env} pantheon session primary-key-finalize to clean up old data, or run terminus wp {site_name}.{env} pantheon session primary-key-revert if there were issues.', 'wp-native-php-sessions' ), 'log' );
```

Comment:
Cannot assume Pantheon environment.

```suggestion
		$this->safe_output( __( 'Operation complete, please verify that your site is working as expected. When ready, run wp pantheon session primary-key-finalize to clean up old data, or run wp pantheon session primary-key-revert if there were issues.', 'wp-native-php-sessions' ), 'log' );
```